src/tox/config/source/toml.py

```diff
@@ -0,0 +1,140 @@
+"""Support for TOML config sources.
+
+This is experimental API! Expect things to be broken.
+"""
+
+from __future__ import annotations
+
+from collections import defaultdict
+from itertools import chain
+from typing import TYPE_CHECKING, Iterable, Iterator
+
+import tomllib
+
+from tox.config.loader.ini.factor import find_envs
+from tox.config.loader.memory import MemoryLoader
+from tox.config.loader.section import Section
+
+from .api import Source
+from .ini_section import CORE, PKG_ENV_PREFIX, TEST_ENV_PREFIX, IniSection
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from tox.config.loader.api import OverrideMap
+    from tox.config.sets import ConfigSet
+
+
+class TomlSource(Source):
+    """Configuration sourced from a toml file (such as tox.toml).
```

Comment:
I do not want a dedicated toml, but should be inside pyproject.toml.

Reply:
I'd prefer to have an option of separate TOML config file as well. If you don't like a dedicated file such as `tox.toml`, what about allowing to specify configuration file in env variable, such as `TOX_CONFIG`?

Reply:
I don't like the environment idea 😃